onnx_xla/onnx_xla_interface.cc

```diff
@@ -45,108 +61,118 @@ struct BackendControl {
 //TODO: Determining # of CPU, GPU, TPU devices and return
 ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
     onnxGetBackendIDs)(onnxBackendID *backendIDs, size_t *numBackends) {
-  try {
-    *backendIDs = (onnxBackendID)(new OnnxXlaBackendID());
+  ONNXIFI_TRY_CATCH(([&] {
+    *numBackends = 0;
+    *backendIDs = reinterpret_cast<onnxBackendID>(new OnnxXlaBackendID());
     *numBackends = 1;
     return ONNXIFI_STATUS_SUCCESS;
-  }
-  ONNXIFI_CATCH_EXCPETION();
+  }))
 }
 
 //Free memory for given backend ID
 ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI
 ONNXIFI_SYMBOL_NAME(onnxReleaseBackendID)(onnxBackendID backendID) {
-  try {
-    auto *backend_id = reinterpret_cast<OnnxXlaBackendID *>(backendID);
-    if (!backend_id) {
+  ONNXIFI_TRY_CATCH(([&] {
+    if (!backendID) {
       return ONNXIFI_STATUS_INVALID_ID;
     }
+    auto *backend_id = reinterpret_cast<OnnxXlaBackendID *>(backendID);
     delete backend_id;
     return ONNXIFI_STATUS_SUCCESS;
-  }
-  ONNXIFI_CATCH_EXCPETION();
+  }))
 }
 
 
 //Returning info for given ID
-//TODO: Expand for different IDs and fill in commented infoType parts
+//TODO: Make sure this information is correct
+//TODO: Expand for different IDs (TPU/GPU) in the future
 ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
     onnxGetBackendInfo)(onnxBackendID backendID, onnxBackendInfo infoType,
                         void *infoValue, size_t *infoValueSize) {
-  try {
+  ONNXIFI_TRY_CATCH(([&] {
+
+    auto SET_STRING = [&](const char* str)  {
+                        strncpy((char *)(infoValue), str, *infoValueSize);
```

Comment:
`onnxGetBackendInfo` returns "all or nothing": if the buffer is insufficient, it must store required buffer size in `*infoValueSize` and return `ONNXIFI_STATUS_FALLBACK`, but not store anything in `infoValue`

Reply:
good point - added check for valid backendID and check for invalid infoValue/insufficient *infoValueXize. Should now return the required buffer size. @yinghai - this should be added to the trt PR as well?